code/modules/atmospherics/machinery/pipes/smart.dm

```diff
@@ -11,16 +11,21 @@ GLOBAL_LIST_INIT(atmos_components, typecacheof(list(/obj/machinery/atmospherics)
 	construction_type = /obj/item/pipe/quaternary
 	pipe_state = "manifold4w"
 	connection_num = 0
+	var/list/connections
 
 /obj/machinery/atmospherics/pipe/smart/update_pipe_icon()
 	icon = 'icons/obj/atmospherics/pipes/pipes_bitmask.dmi'
 	var/bitfield = NONE
 	var/bits = 0
+	connections = 0
+	connection_num = NONE
 	for(var/i in 1 to device_type)
 		if(!nodes[i])
 			continue
 		var/obj/machinery/atmospherics/node = nodes[i]
 		var/connected_dir = get_dir(src, node)
+		connections |= connected_dir
+		connection_num += 1
```

Comment:
was connection_num broken?

Reply:
Tested it after my last commit and it works.

Reply:
Oh it was totally unused before now. what the fuck

Reply:
Before I accidentally made connection_num start as NONE instead of 0. It's used for the switch statement.